Core/src/org/sleuthkit/autopsy/contentviewers/HtmlViewer.java

```diff
@@ -65,14 +68,35 @@ private String getHtmlText(AbstractFile abstractFile) {
             int fileSize = (int) abstractFile.getSize();
             byte[] buffer = new byte[fileSize];
             abstractFile.read(buffer, 0, fileSize);
-            return new String(buffer);
-        } catch (TskCoreException ex) {
+            String encoding = determineEncoding(buffer);
+            if (encoding != null) {
+                return new String(buffer, encoding);
+            } else {
+                return new String(buffer);
+            }
+        } catch (TskCoreException | IOException ex) {
             logger.log(Level.SEVERE, String.format("Unable to read from file '%s' (id=%d).",
                     abstractFile.getName(), abstractFile.getId()), ex);
             return String.format("<p>%s</p>", Bundle.HtmlViewer_file_error());
         }
     }
 
+    /**
+     * This method will try and determine the encoding of the html file based on its contents
+     * 
+     * @param buffer byte array of the html file to check
+     * 
+     * @return encoding type, null if encoding could not be determined
+     */
+    private String determineEncoding(byte[] buffer) throws IOException {
+  
```

Comment:
I don't think any of the methods you use here throw an IOException so that's not needed.
Once you get rid of that you should notice that you will again have the possibility of an UnsupportedEncodingException in getHtmlText() above.

Also, detector.getDetect() can return null so you need to handle that before calling CharsetMatch.getName().

Personally, I would do away with this method and do the following in getHtmlText() but I'll leave it up to you.

            CharsetMatch match = new CharsetDetector().setText(buffer).detect();
            if (match != null) {
                return new String(buffer, match.getName());
            } else {
                return new String(buffer);
            }

Reply:
Changed code per your suggestion, much cleaner.

Reply:
I like it a lot :-)